packages/selected-network-controller/tests/SelectedNetworkController.test.ts

```diff
@@ -0,0 +1,138 @@
+  });
+
+  it('can set the networkClientId for the metamask domain specifically', () => {
+    const options: SelectedNetworkControllerOptions = {
+      messenger: buildSelectedNetworkControllerMessenger(), // Mock the messenger
+    };
+    const controller = new SelectedNetworkController(options);
+    const networkClientId = 'network2';
+    controller.setNetworkClientIdForMetamask(networkClientId);
+    expect(controller.state.domains.metamask).toBe(networkClientId);
+  });
+
+  it('can reset the state / selections to the default state', () => {
+    const options: SelectedNetworkControllerOptions = {
+      messenger: buildSelectedNetworkControllerMessenger(), // Mock the messenger
+    };
+    const controller = new SelectedNetworkController(options);
+    controller.setNetworkClientIdForDomain('example.com', 'network1');
+    controller.setNetworkClientIdForDomain('test.com', 'network2');
+    controller.setNetworkClientIdForMetamask('network3');
+    controller.resetState();
+    expect(controller.state).toStrictEqual({
+      domains: {},
+      perDomainNetwork: false,
+    });
+  });
+
+  describe('getNetworkClientIdForDomain', () => {
+    it('gives the metamask domain when the perDomainNetwork option is false (default)', () => {
+      const options: SelectedNetworkControllerOptions = {
+        messenger: buildSelectedNetworkControllerMessenger(), // Mock the messenger
+      };
+      const controller = new SelectedNetworkController(options);
+      const networkClientId = 'network4';
+      controller.setNetworkClientIdForMetamask(networkClientId);
+      const result = controller.getNetworkClientIdForDomain('example.com');
+      expect(result).toBe(networkClientId);
+    });
+
+    it('when the perDomainNetwork feature flag is on, it returns items other than the metamask domain', () => {
+      const options: SelectedNetworkControllerOptions = {
+        messenger: buildSelectedNetworkControllerMessenger(), // Mock the messenger
+      };
+      const controller = new SelectedNetworkController(options);
+      controller.state.perDomainNetwork = true;
+      const networkClientId1 = 'network5';
+      const networkClientId2 = 'network6';
+      controller.setNetworkClientIdForDomain('example.com', networkClientId1);
+      controller.setNetworkClientIdForDomain('test.com', networkClientId2);
+      const result1 = controller.getNetworkClientIdForDomain('example.com');
+      const result2 = controller.getNetworkClientIdForDomain('test.com');
+      expect(result1).toBe(networkClientId1);
+      expect(result2).toBe(networkClientId2);
+    });
+  });
+
+  it('updates the networkClientId for the metamask domain when the networkControllers selectedNetworkClientId changes', () => {
+    const mockMessagingSystem = {
+      registerActionHandler: jest.fn(),
+      subscribe: jest.fn(),
+      publish: () => jest.fn(),
+    };
+    const options: SelectedNetworkControllerOptions = {
+      messenger: mockMessagingSystem as any,
```

Comment:
I think we should do our best to avoid using `any` even in tests. Is there any reason why we can't use a real messenger object here?

Reply:
done! That said, I do quite like the simplicity of using mocks instead of real messenger. Not having to deal with the typing there is nice (when you have tests that are already covering the typings of the messenger).